### DISCUSSION TITLE
Can I get the nine predefined temporal predicates' lustre node code ?

### DISCUSSION THREAD

**Discussion by Atmanuclear** [NASA-SW-VnV/fret#52] [Q&A] [answered]
I noticed that in the [Writing Requirements in FRET](https://github.com/NASA-SW-VnV/fret/blob/master/fret-electron/docs/_media/user-interface/writingReqs.md) you say there are some predefined temporal predicates,as I want to transfer a FRETish set into one Lustre Controller, I found that in the cocospec which only predefine preInt preReal and preBool, I want to know more about persisted and occurred or other predefined temporal Lustre node! Thanks very much !

**Comment by tpressburger** [comment, marked as answer]
There are Lustre nodes for preReal and preInt, but not
for the other temporal predicates. The other temporal predicates are defined
by rewrite rules listed in the variables pastTemporalConditions and
futureTemporalConditions in fret-electron/support/xform.js. The identifiers
$Left$ and $Right$ will be instantiated with the predicates defining the
left and right endpoints of the scope interval.  Actually there is a Lustre
node for preBool as well as a rewrite rule; we will think about removing the
Lustre node.
